packages/js/src/server/integrations/unhandled_rejection.ts

```diff
@@ -19,18 +19,17 @@ export default function (): Types.Plugin {
         return
       }
 
-      const hasOtherListeners = hasUnhandledRejectionListeners();
       process.on('unhandledRejection', function honeybadgerUnhandledRejectionListener(reason, _promise) {
         if (!client.config.enableUnhandledRejection) {
-          if (!hasOtherListeners) {
+          if (!hasOtherUnhandledRejectionListeners()) {
```

Comment:
Whoops, sorry, another bug. This is being called inside `if (!client.config.enableUnhandledRejection) {`, which means checking that `listeners.length > 1` might not be accurate. I'm not sure why we even register a handler when that is turned off. Isn't that counter to user's expectations? IMO, we should invert the conditions on the previous lines, leading to:

```js
      if (!client.config.enableUnhandledRejection) {
        process.on('unhandledRejection', function honeybadgerUnhandledRejectionListener(reason, _promise) {
          if (!hasOtherUnhandledRejectionListeners()) {
```

Also not your code, but definitely something I think we can do better (and maybe might be worth packaging it into the major at this point.)

Reply:
> This is being called inside if (!client.config.enableUnhandledRejection) {, which means checking that listeners.length > 1 might not be accurate

Why is it not accurate? It's inside our registered handler, which means we know there's at least one registered handler (this one).

By the way, I had the same question when I first read this code. Here's how I understand it (assuming this is intended behaviour):

First, on HB init, check if unhandled rejections should be reported to HB.
- If not, return.
- If yes, proceed to register handler:
  - Handler called, **check if config is still enabled. Why? Because we allow modifying the config anytime (`Honeybadger.configure`)**
    - If not enabled, check if there are any other handlers before deciding whether to exit or not.
    - If enabled, proceed to report to HB.

Reply:
Oh, okay. You're right. I missed the first check, and I thought that was what this inner check was trying to do.